digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/ViiteApi.scala

```diff
@@ -207,9 +209,12 @@ class ViiteApi(val roadLinkService: RoadLinkService, val vVHClient: VVHClient,
     val user = userProvider.getCurrentUser()
     val roadAddressProject = ProjectConverter.toRoadAddressProject(project, user)
     try {
-      val (projectSaved, addr, info, success) = projectService.createRoadLinkProject(roadAddressProject)
-      Map("project" -> roadAddressProjectToApi(projectSaved), "projectAddresses" -> addr, "formInfo" -> info,
-        "success" -> success)
+      val projectSaved = projectService.createRoadLinkProject(roadAddressProject)
+      val fetched = projectService.getRoadAddressSingleProject(projectSaved.id).get
+      val firstAddress: Map[String, Any] =
+        fetched.reservedParts.find(_.startingLinkId.nonEmpty).map(p => "projectAddresses" -> p.startingLinkId.get).toMap
+      Map("project" -> roadAddressProjectToApi(fetched), "formInfo" ->
+        fetched.reservedParts.map(reservedRoadPartToApi), "success" -> true) ++ firstAddress
```

Comment:
do we need "trPreview" as in line 239?

Reply:
I think it's used to refresh the change table view. Or should be.

Reply:
Well, not after saving the project. I'll remove that change table and it'll be faster, too.